src/seedu/addressbook/data/tagging/UniqueTaggingList.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.tagging;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import seedu.addressbook.data.tag.Tag;
+
+public class UniqueTaggingList {
```

Comment:
AddressBook should be the one that takes care of the tagging, according to the class diagram.